docs/source/design/dataset_argument.rst

```diff
@@ -0,0 +1,58 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_dataset_argument:
+
+********************
+``dataset`` argument
+********************
+
+All commands which operate on datasets have a ``dataset`` argument (``-d`` or
+``--dataset`` for the command line API) to identify a single dataset as the
+context of an operation.
+
+
+Impact on relative path resolution
+==================================
+
+With one exception, the nature of a provided ``dataset`` argument does **not**
+impact the interpretation of relative paths. Relative paths are always considered
+to be relative to the process working directory.
+
+The one exception to this rule is passing a ``Dataset`` object instance as
+``dataset`` argument value in the Python API. In this, and only this, case, a
+relative path is interpreted as relative to the root of the respective dataset.
+
+
+Special values
+==============
+
+There are some pre-defined "shortcut" values for dataset arguments:
+
+``^``
+   Represents to the topmost superdataset that contains the dataset the current
+   directory is part of.
+``^.``
+   Represents the root directory of the dataset the current directory is part of.
+``///``
+   Represents the "default" dataset located under `$HOME/datalad/`.
+
+
+Use cases
+=========
+
+Save modification in superdataset hierarchy
+-------------------------------------------
+
+Sometimes it is convenient to work only in the context of a subdataset.
+Executing a ``datalad save <subdataset content>`` will record changes to the
+subdataset, but will leave existing superdatasets dirty, as the subdataset
```

Comment:
While I generally agree (and see the technical reason, of course), I wonder whether this approach really is, what we should do. "Sometimes it is convenient to work only in the context of a subdataset." True. And: Given the aim for a "mono-repo experience", it makes sense to *only sometimes* "work only in the context of a subdataset". However, since the context by default is derived from a path argument (as shown here in `datalad save <subdataset content>`) and pretty much all commands do operate on paths, the mono-repo view is *not* the default. Quite the opposite: In order to know what a "simple" (as in "no options specified") invocation like this will do, I need to be aware of the subdataset boundaries. Not sure about a solution, but for now I wonder whether we agree there's an issue. May be it's just me ...

Reply:
I am not sure what you are asking for? Remove the example? Change how `save` works?

Reply:
As far as documenting the status quo goes the only asking is my comment further up - pointing out how PWD and a possible `path` argument contribute to determining the context.

Other than that: Wondering whether we are satisfied with how `-d` works (it's not about `save`). The suggestion would be to take `path` out of the equation and make the context always `-d` with default `.`. This would be a breaking change - yes (But easy in implementation, I think). ATM I think it would be much more intuitive, though.